ARMeilleure/Instructions/InstEmitSimdCmp.cs

```diff
@@ -538,9 +538,12 @@ private static void EmitFcmpOrFcmpe(ArmEmitterContext context, bool signalNaNs)
 
                     context.BranchIfFalse(lblNaN, isOrdered);
 
-                    Operand cf = context.AddIntrinsicInt(Intrinsic.X86Comissge, n, m);
-                    Operand zf = context.AddIntrinsicInt(Intrinsic.X86Comisseq, n, m);
-                    Operand nf = context.AddIntrinsicInt(Intrinsic.X86Comisslt, n, m);
+                    Operand nCopy = context.Copy(n);
+                    Operand mCopy = cmpWithZero ? null : context.Copy(m);
+
+                    Operand cf = context.AddIntrinsicInt(Intrinsic.X86Comissge, nCopy, cmpWithZero ? context.VectorZero() : mCopy);
+                    Operand zf = context.AddIntrinsicInt(Intrinsic.X86Comisseq, nCopy, cmpWithZero ? context.VectorZero() : mCopy);
+                    Operand nf = context.AddIntrinsicInt(Intrinsic.X86Comisslt, nCopy, cmpWithZero ? context.VectorZero() : mCopy);
```

Comment:
```suggestion
                    Operand nCopy = context.Copy(n);
                    Operand mCopy = cmpWithZero ? m : context.Copy(m);

                    Operand cf = context.AddIntrinsicInt(Intrinsic.X86Comissge, nCopy, mCopy);
                    Operand zf = context.AddIntrinsicInt(Intrinsic.X86Comisseq, nCopy, mCopy);
                    Operand nf = context.AddIntrinsicInt(Intrinsic.X86Comisslt, nCopy, mCopy);
```

I believe doing this instead might improve the codegen further for the zero case, because it removes some of the redundant `VectorZero` operations. Likewise for the `op.Size == 1` path.

Reply:
Those repeated `VectorZero` are wanted, otherwise `VectorZero` is simply spilled; the goal here is to avoid or reduce interactions with the stack.

Reply:
Thanks, makes sense. 👍

Reply:
FWIW I investigated this further, and some of the `Fill`/`VectorZero` in LCQ can be avoided if
```patch
- Operand mCopy = cmpWithZero ? m : context.Copy(m);
+ Operand mCopy = cmpWithZero ? context.VectorZero() : context.Copy(m);
```
is used instead.

```patch
 .L5:
   vmovdqu xmm0,oword [rsp+0x5c]
   vmovdqu xmm4,xmm0
   vxorps xmm5,xmm5,xmm5
   vcomiss xmm4,xmm5
   setnc bpl
   movzx ebp,bpl
-  vxorps xmm5,xmm5,xmm5
   vcomiss xmm4,xmm5
   setz sil
   movzx esi,sil
-  vxorps xmm5,xmm5,xmm5
   vcomiss xmm4,xmm5
   setc dil
   movzx edi,dil
   xor eax,eax
```
And turns out the LSRA can handle the original suggestion without any spills, since the interval of `m` spans across the 2 blocks without any holes. But I suppose it is a fair tradeoff between the CQ modes.

Reply:
I think keeping the zero value on a register is better, otherwise it will do the zero-ing pattern (with `XORPS`) every time `VectorZero` is used. The only cases where it's not better, is when there's high register pressure that could cause the zero value to be spilled and reloaded, but I think on HighCq, the register allocator should be smart enough to not spill/reload the zero value everytime for such a short interval.